dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -190,4 +212,79 @@ public String asJson(final PageView pageView) throws JsonProcessingException {
         return objectWriter.writeValueAsString(pageView);
     }
 
+    public Template saveTemplate(final User user, String pageId, final PageForm pageForm)
+            throws BadRequestException, DotDataException, DotSecurityException, IOException {
+
+        final Contentlet page = this.contentletAPI.findContentletByIdentifier(pageId, false,
+                langAPI.getDefaultLanguage().getId(), user, false);
+
+        if (page == null) {
+            throw new NotFoundException("An error occurred when proccessing the JSON request");
+        }
+
+        try {
+            Template templateSaved = this.saveTemplate(page, user, pageForm);
+
+            String templateId = page.getStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD);
+
+            if (!templateId.equals( templateSaved.getIdentifier() )) {
+                page.setStringProperty(Contentlet.INODE_KEY, null);
+                page.setStringProperty(HTMLPageAssetAPI.TEMPLATE_FIELD, templateSaved.getIdentifier());
+                this.contentletAPI.checkin(page, user, false);
+            }
+
+            return templateSaved;
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)